### DISCUSSION TITLE
Gradle: Resource missing for dependency when using GITHUB_TOKEN

### DISCUSSION THREAD

**Discussion by mindhaq** [community/community#25513] [Packages]
<p>I have a project residing in two GH repositories. There is library in common, and the application is using that library.</p>
<p>I have a successfully published package of the library in GitHub Packages.</p>
<p>I have setup the application project in a way that if I build it locally, it will download the library using a personal access token.</p>
<p>build.gradle:</p>
<pre><code>repositories {
    mavenLocal()
    mavenCentral()
    maven {
        name = "GitHubPackages-common"
        url = "https://maven.pkg.github.com/OWNER/rr-common"
        credentials {
            username = System.getenv("GITHUB_USERNAME")
            password = System.getenv("GITHUB_TOKEN")
        }
    }
}
</code></pre>
<p>In the GH action for building the application, I use the GITHUB_TOKEN, and when I print that out in the build script, it seems to be passed along correctly.</p>
<p>build.yml:</p>
<pre><code>- name: Build with Gradle
      env:
        GITHUB_USERNAME: "OWNER"
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      run: ./gradlew --stacktrace --warning-mode all --info clean test bootJar
</code></pre>
<p>However, the build fails because the library cannot be downloaded. First trying at maven central (unsuccessfully as expected) and then trying at github, and also failing.</p>
<pre><code>Resource missing. [HTTP GET: https://repo.maven.apache.org/maven2/de/OWNER/common/1.0.3/common-1.0.3.pom]
Resource missing. [HTTP HEAD: https://repo.maven.apache.org/maven2/de/OWNER/rr-common/1.0.3/common-1.0.3.jar]
Resource missing. [HTTP GET: https://maven.pkg.github.com/OWNER/common/de/OWNER/common/1.0.3/common-1.0.3.pom]
Resource missing. [HTTP HEAD: https://maven.pkg.github.com/OWNER/common/de/OWNER/rr-common/1.0.3/common-1.0.3.jar]
</code></pre>
<p>The failing URLs work fine when entered into a browser (after authentication).</p>
<p>A working workaround for me was now to create a user-specific API Token, put that into the projects secrets and use that from within the build action. And that works! But isn’t that exactly what GITHUB_TOKEN should be used for?</p>

**Comment by BrightRan** [comment, marked as answer]
<p>The following note may be able to answer your doubt.<br>
<strong>NOTE: The permissions of GITHUB_TOKEN are limited to the repository that contains your workflow. If you need a token that requires permissions that aren’t available in the GITHUB_TOKEN, you can create a personal access token and set it as a secret in your repository.</strong><br>
As the note mentioned, if your common library is not published on the same GitHub repository where the workflow is hosted, it is not able to download the library via the GITHUB_TOKEN in the workflow. So you need to create a new PAT to get more permission scopes.<br>
More detailed info about GITHUB_TOKEN, you can reference here: <a href="https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token</a></p>

**Comment by mindhaq** [comment]
<p>Understood. I was probably hoping to get the same developer experience as I get when using NPM packages.</p>
<p>I opened another posting for that, as it has nothing to do with GitHub actions.</p>
<p><a href="https://github.community/t/github-package-registry-as-maven-repo-with-many-private-libraries/14538">https://github.community/t5/GitHub-API-Development-and/GitHub-package-registry-as-Maven-repo-with-many-private/m-p/41488#M3734</a></p>

**Comment by hfhbd** [comment]
But it should be possible to use `packages: read` with the GitHub token provided by Actions to fetch the published maven artifacts to the same repo, shouldn't it?
